packages/context/src/binding-config.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingAddress, BindingKey} from './binding-key';
+import {Context} from './context';
+import {ResolutionOptions} from './resolution-session';
+import {ValueOrPromise} from './value-promise';
+
+/**
+ * Interface for configuration resolver
+ */
+export interface ConfigurationResolver {
+  /**
+   * Resolve the configuration value for a given binding key and config property
+   * path
+   * @param key Binding key
+   * @param configPath Config property path
+   * @param resolutionOptions Resolution options
+   */
+  getConfigAsValueOrPromise<ConfigValueType>(
+    key: BindingAddress<unknown>,
+    configPath?: string,
+    resolutionOptions?: ResolutionOptions,
+  ): ValueOrPromise<ConfigValueType | undefined>;
+}
+
+/**
+ * Resolver for configurations of bindings
+ */
+export class DefaultConfigurationResolver implements ConfigurationResolver {
+  constructor(public readonly context: Context) {}
+
+  /**
+   * Resolve config for the binding key
+   *
+   * @param key Binding key
+   * @param configPath Property path for the option. For example, `x.y`
+   * requests for `<config>.x.y`. If not set, the `config` object will be
+   * returned.
+   * @param resolutionOptions Options for the resolution.
+   * - optional: if not set or set to `true`, `undefined` will be returned if
+   * no corresponding value is found. Otherwise, an error will be thrown.
```

Comment:
I think this detailed tsdoc comment should be moved to `ConfigurationResolver` description, so that people implementing `ConfigurationResolver` better understand what to expect in the arguments.